### ISSUE TITLE
[v3.0.2] Plugin.prototype.readFile this.outputs[file] undefined because of OS path separator

### ISSUE THREAD

**Issue by klmdb** [webpack-contrib/karma-webpack#349]
going from 2.x to 3.x, the code became dependent on os path separator:

2.0.13
```javascript
middleware.fileSystem.readFile(path.join(os.tmpdir(), '_karma_webpack_', String(idx), file.replace(/\\/g, '/')), callback)
//...
var fileContents = middleware.fileSystem.readFileSync(path.join(os.tmpdir(), '_karma_webpack_', file.replace(/\\/g, '/')))
```

3.0.2
```javascript
middleware.fileSystem.readFile(path.join(os.tmpdir(), '_karma_webpack_', String(idx), this.outputs[file]), callback)
//...
var fileContents = middleware.fileSystem.readFileSync(path.join(os.tmpdir(), '_karma_webpack_', this.outputs[file]))
```

`this.outputs[file]` should be `this.outputs[file.replace(/\\/g, '/')]`

**Comment by michael-ciniawsky** [comment]
Could you try with `path.posix.join` or a small helper 

``js
const normalize = (file) => file.split(path.sep).join('/')
...
... this.outputs[normalize(file)] ...
``` 

I'm assuming your are on windows, right ? Otherwise please elaborate in more detail what doesn't work please (Actual/Expected)

**Comment by rikuw** [comment]
Running karma-webpack on Windows results in the following error message on 3.0.2:

`ERROR [karma]: [TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined`

Running on 2.0.0 works.
The problem seems to be that `this.outputs` is indexed with paths using forward slash separators while backslash separators are not replaced in the indexing string. This means every reference to files in different directories on windows-machines throws an error.

**Comment by Javi** [comment]
Can confirm, just got bit by this on Windows and making paths like `[\`folder${path.delimiter}file.js\`]` makes it work.

**Comment by nomve** [comment]
I don't run Windows but still have the same issue

https://github.com/webpack-contrib/karma-webpack/blob/d2f5a538d864e7b1f4bb431324de67be1ccbb51d/src/karma-webpack.js#L238

`this.outputs` is just an empty `{}`. the only place I see it get written is here
https://github.com/webpack-contrib/karma-webpack/blob/d2f5a538d864e7b1f4bb431324de67be1ccbb51d/src/karma-webpack.js#L131

but this runs later.

**Comment by nomve** [comment]
downgrading to `3.0.0` also works for me

**Comment by michael-ciniawsky** [comment]
Fixed by #353, released in [`v3.0.3`](https://github.com/webpack-contrib/karma-webpack/releases/tag/v3.0.3) 🎉
